File: CodeWars/.js/8kyu-Welcome-to-the-city.js
// Create a method sayHello/say_hello/SayHello that takes as input a name, city, and state to welcome a person. Note that name will be an array consisting of one or more values that should be joined together with one space between each, and the length of the name array in test cases will vary.

// Example:

// sayHello(['John', 'Smith'], 'Phoenix', 'Arizona')
// This example will return the string Hello, John Smith! Welcome to Phoenix, Arizona!

const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Sample Tests", () => {
  it("Should pass Sample Tests", () => {
    assert.strictEqual(sayHello(['John', 'Smith'], 'Phoenix', 'Arizona'), 'Hello, John Smith! Welcome to Phoenix, Arizona!')
    assert.strictEqual(sayHello(['Franklin','Delano','Roosevelt'], 'Chicago', 'Illinois'), 'Hello, Franklin Delano Roosevelt! Welcome to Chicago, Illinois!')
    assert.strictEqual(sayHello(['Wallace','Russel','Osbourne'],'Albany','New York'), 'Hello, Wallace Russel Osbourne! Welcome to Albany, New York!')
  });
});

// my code

// 
// @param: three string inputs for name, city, state
// @return: welcome string
// 

function sayHello( name, city, state ){
    return `Hello, ${name.join(' ')}! Welcome to ${city}, ${state}!`
  }

File: CodeWars/.cs/8kyu-WelcomeToTheCity.cs
// Create a method that takes as input a name, city, and state to welcome a person. Note that name will be an array consisting of one or more values that should be joined together with one space between each, and the length of the name array in test cases will vary.

// Example:

// ['John', 'Smith'], 'Phoenix', 'Arizona'
// This example will return the string Hello, John Smith! Welcome to Phoenix, Arizona!

using System;

namespace Solution {
  using NUnit.Framework;
  using System;

  [TestFixture]
  public class SolutionTest
  {
    [Test]
    public void SampleTest()
    {
      Assert.AreEqual("Hello, John Smith! Welcome to Phoenix, Arizona!", Kata.SayHello(new string[] {"John", "Smith"}, "Phoenix", "Arizona"));
    }
  }
}

// my code

public class Kata
{
  public static string SayHello(string[] name, string city, string state)
  {
    string fullName = string.Join(' ', name);
    return $"Hello, {fullName}! Welcome to {city}, {state}!";    
  }
}